### ISSUE TITLE
Support Weblate Flags in Gettext Files via Extracted Comments

### ISSUE THREAD

**Issue by anubeon** [WeblateOrg/weblate#11655]
### Describe the problem

We use GNU gettext for translating strings for embedded systems, with our software engineers providing human-readable cues for things like maximum character length via the extracted comments feature (example below).

```
#.
#. STARTUP TYPEWRITER TEXT
#. ------------------------------------------------------------------
#. EEPROM BLANK (Maxlen=12)
msgid "BLANK"
msgstr "VIDE"

#. (DO NOT TRANSLATE!)
msgid "EEPROM"
msgstr ""
```

For our use case, this could potentially be a far more efficient means of setting [weblate-flags](https://docs.weblate.org/en/latest/admin/checks.html) such as `max-length:N` than manually setting said flags via Weblate's UI. Presently, however, only certain XML based formats support storage and synchronisation of [weblate-flags](https://docs.weblate.org/en/latest/admin/checks.html).

### Describe the solution you would like

We suggest supporting storage and synchronisation of [weblate-flags](https://docs.weblate.org/en/latest/admin/checks.html) within gettext's extracted comments, in a similar vain to the non-standard 

As the extracted comments feature of GNU gettext is designed to allow programmers to provide cues to translators, this would seem the ideal means of conveying machine-readable instructions to a translation tool such as Weblate.

### Example:
```
#.
#. STARTUP TYPEWRITER TEXT
#. ------------------------------------------------------------------
#. EEPROM BLANK weblate-flags="max-length:12, priority:50"
msgid "BLANK"
msgstr "VIDE"

#. weblate-flags="read-only"
msgid "EEPROM"
msgstr ""

```

In the example above, [weblate-flags](https://docs.weblate.org/en/latest/admin/checks.html) are:

1. Appended to the end of the extracted comments.
2. Signified via the keyword/prefix `weblate-flags=`.
3. Encapsulated by double quotes.
4. Comma-delimited.

These would seem like reasonable conventions to follow?

**Comment by nijel** [comment]
Weblate does extract flags from Gettext type comments:

```po
#: weblate/addons/forms.py:541
#, python-format
msgid "Could not parse CSS selector: %s"
msgstr "Nepodařilo se zpracovat CSS selektor: %s"
```

Will get `python-format` flag. The issue with that is that there is no way to generate custom flags using gettext and I think that gettext tools will complain about unknown flags.

**Comment by anubeon** [comment]
> It makes sense to support extracting Weblate flags from the comments, but the syntax has to be carefully designed. For example, the flags itself can contain quotes, so using quotes to delimit them doesn't sound like a good idea.

@nijel Of course, I'm not wedded to the use of quotes as delimited. Can you see any issues with using curly brackets? Again, I'm not wedded to this.

```
#.
#. STARTUP TYPEWRITER TEXT
#. ------------------------------------------------------------------
#. EEPROM BLANK weblate-flags{ max-length:12, priority:50" }
msgid "BLANK"
msgstr "VIDE"

#. weblate-flags{ "read-only }
msgid "EEPROM"
msgstr ""
```

> Please try to clarify the use case or consider proposing something more generic to make it useful to more users.

The use case would be identical to that which (presumably) warranted the implementation of [weblate-flags](https://docs.weblate.org/en/latest/admin/checks.html) in XML based formats such as XLIFF. 

Of course one could argue "Why not just use XLIFF?", but there are presumably plenty of projects out there which rely on gettext, and which aren't in a position to migrate, that might find the ability to communicate [weblate-flags](https://docs.weblate.org/en/latest/admin/checks.html) such as `read-only`, and `min-length:N` useful?

**Comment by nijel** [comment]
ITS content ends up in the extracted comment, this is a snipped from a POT file @nicorikken linked in #14722:

```po
#. (itstool) path: info/title
#. (itstool) path: cover/bridgehead
#. Ampersand character & is escaped for XML source format as &amp;
#.
#. ignore-same
#: Ada_Zangemann.dbk:263
#: Ada_Zangemann.dbk:307
#: Ada_Zangemann.dbk:497
msgid "Ada &amp; Zangemann"
msgstr ""
```

**Comment by nijel** [comment]
GNU gettext is going to support custom sticky flags in PO files (see https://lists.gnu.org/archive/html/bug-gettext/2025-06/msg00018.html) and I believe this is a better approach to this. Let's see how it evolves.

**Comment by mahlzahn** [comment]
### Workaround

As long as gettext doesn’t have it yet, I myself implemented source code comments (here in Python) like

```py
error = 'an error'
# Open Document Warning
# weblate-flags: priority:50
_('Failed to open document from path: {error}').format(error=error)
```

with a custom comman called after the POT and PO files generation

```sh
perl -0777 -i -pe 's/#\. weblate-flags:\s*([^\n]+)\n((?:#.*\n)*)/\2#, \1\n/g;s/(#,\s.*)\n#,\s(.*)/\1, \2/g' po/*pot po/*.po
```

which leads to the following in the POT file:

```gettext
#. Open Document Warning
#: src/document.py:44
#, python-brace-format, priority:50
msgid "Failed to open document from path: {error}"
msgstr ""
```

### Implementation in Weblate (if not gettext gets it)

In case, gettext doesn’t get it I’d vote for a very simple format

```gettext
#. weblate-flags: max-length:12, priority:50
```

to be directly supported by Weblate. If a comment (`#. `) starts with `#. weblate-flags:`, interpret the comma-separated list after that as additional flags.

**Comment by mahlzahn** [comment]
> which leads to the following in the POT file:
> 
> ```
> #. Open Document Warning
> #: src/document.py:44
> #, python-brace-format, priority:50
> msgid "Failed to open document from path: {error}"
> msgstr ""
> ```

@nijel But unfortunately, it isn't picked up by Weblate, in the single languages: https://translate.codeberg.org/browse/censor/-/ms/?q=priority%3A50.

Edit: The English original (from the POT file) has it: https://translate.codeberg.org/browse/censor/-/en/?q=priority%3A50. I use Weblate's extension to update PO files https://docs.weblate.org/en/weblate-2026.7.1/admin/addons.html#addon-weblate-gettext-msgmerge.

Edit2: And of course, it is just calling `msgmerge` and this seems to drop custom flags.

Edit3: If you look now again at https://translate.codeberg.org/browse/censor/-/ms/?q=priority%3A50, there are couple of `priority:50` strings, but I updated them using bulk editing, searching for `{error} OR "Failed to" OR found`. Not a very satisfying workaround.
